server/start.go

```diff
@@ -310,30 +313,46 @@ func startInProcess(ctx *server.Context, clientCtx client.Context, appCreator ty
 	}
 
 	genDocProvider := node.DefaultGenesisDocProviderFunc(cfg)
-	tmNode, err := node.NewNode(
-		cfg,
-		pvm.LoadOrGenFilePV(cfg.PrivValidatorKeyFile(), cfg.PrivValidatorStateFile()),
-		nodeKey,
-		proxy.NewLocalClientCreator(app),
-		genDocProvider,
-		node.DefaultDBProvider,
-		node.DefaultMetricsProvider(cfg.Instrumentation),
-		ctx.Logger.With("server", "node"),
+	var (
+		tmNode *node.Node
+		modes  = strings.Split(ctx.Viper.GetString(srvflags.QueryOnly), ",")
 	)
-	if err != nil {
-		logger.Error("failed init node", "error", err.Error())
-		return err
-	}
+	gRPCQueryMode := slices.Contains(modes, "grpc")
+	jsonRPCMode := slices.Contains(modes, "jsonrpc")
+	if gRPCQueryMode || jsonRPCMode {
+		ctx.Logger.Info(fmt.Sprintf("starting node in %s query only mode; Tendermint is disabled", modes))
+		if gRPCQueryMode {
+			config.GRPC.Enable = true
+		}
+		if jsonRPCMode {
+			config.JSONRPC.Enable = true
+		}
```

Comment:
```suggestion
```
Think this can be removed, since these both default to `true`. Would make sense that user should select at least one of these to be `true` when `QueryMode` is enabled, can make sure this is true earlier in this function or `StartCmd` if this is a concern.